### ISSUE TITLE
Not working on PopOS?

### ISSUE THREAD

**Issue by CodexHere** [maweki/twitchlive-extension#100]
Hey there, I installed PopOS which runs gnome 42.9, and it appears that there's an error during importing.

The stacktrace doesn't indicate it's necesarially an issue with your code, but I was wondering if it was something you'd be willing to look into and see if it's a bug in the code, or something Gnome broke (wouldn't be surprised if it's gnome):

```
 Unhandled promise rejection. To suppress this warning, add an error handler to your promise chain with .catch() or a try-catch block around your await expression. Stack trace of the failed promise:
                                       run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:186:20
                                       main@resource:///org/gnome/Shell/Extensions/js/main.js:22:13
                                       run@resource:///org/gnome/gjs/modules/script/package.js:206:19
                                       start@resource:///org/gnome/gjs/modules/script/package.js:190:8
                                       @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

**Comment by maweki** [comment]
The follows API was recently deprecated and I've changed it to the new API. Bit I haven't tested the code. Anyways, this code is only in the gnome 44 release.

https://github.com/maweki/twitchlive-extension/commit/7a812b2d3beda8bf1e29ac72838130ebfcfc4748

You could try to apply this patch to your local extension version and try it out.

So I'm not sure it is gnome. It could be our bad error handling of twitch's response.

**Comment by CodexHere** [comment]
Yep, those changes seem to have mostly fixed it. The settings seems to crash frequently. Although it is a minimal change to scope and an ID, so I don't know why it would be crashing now when it never did before.

I think soon enough PopOS will do away with gnome as the default WDE anyway, so I have to figure out what I want to do about it 😅

Either way, this is good enough for now, thanks!
